### PR TITLE
Add prop `preferredSKU` to Gallery

### PR DESCRIPTION
**Pull request by Git-the-Sanz**
#### What problem is this solving?

As the `Gallery` is the one responsible of mapping the catalog products to the summary context, we needed to create a new prop, the closest to the method call as possible

#### How to test it?

[CHEAPEST](https://cheapest--motoroladevelopment.myvtex.com/)
(You will see the cheapest products selected)

#### Screenshots or example usage:
For i.e:
If you pass "PRICE_ASC" the cheapest SKU will be automatically selected in the Gallery

prop
item

#### Related to / Depends on

[Product summary PR](https://github.com/vtex-apps/product-summary/pull/328)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -228,88 +228,100 @@ Structure the `search-result-layout` or the `search-result-layout.customQuery`,
...
+
+| Name            | Value             | Description                                        |
+| --------------- | ----------------- | -------------------------------------------------- |
+| First Available | `FIRST_AVAILABLE` | Selects the first available SKU in stock it finds. |
+| Last Available  | `LAST_AVAILABLE`  | Selects the last available SKU in stock it finds.  |
+| Cheapest        | `PRICE_ASC`       | Selects the cheapest SKU in stock it finds.        |
+| Most Expensive  | `PRICE_DESC`      | Selects the most expensive SKU in stock it finds.  |
+
+⚠️ Keep in mind that the default behaviour for the initially selected SKU, has now changed to a Product (field) specification.  
+If that one doesn't exist, it will use the `preferredSKU` prop. You can read more about it, and how to implement it in [Recipes](https://vtex.io/docs/recipes/all)
```
Comment (victorhmp) [inline docs/README.md:308]: Same as https://github.com/vtex-apps/product-summary/pull/328#discussion_r689766102

File: react/package.json [hunk 2]
```diff
@@ -41,28 +41,28 @@
...
     "vtex.flex-layout": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.flex-layout@0.17.0/public/@types/vtex.flex-layout",
     "vtex.format-currency": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.format-currency@0.3.0/public/@types/vtex.format-currency",
     "vtex.native-types": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.native-types@0.7.5/public/@types/vtex.native-types",
     "vtex.pixel-manager": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.pixel-manager@1.8.0/public/@types/vtex.pixel-manager",
     "vtex.product-list-context": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.product-list-context@0.4.0/public/@types/vtex.product-list-context",
-    "vtex.product-summary": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.product-summary@2.73.0/public/@types/vtex.product-summary",
-    "vtex.render-runtime": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.render-runtime@8.128.4/public/@types/vtex.render-runtime",
+    "vtex.product-summary": "https://specification--motoroladevelopment.myvtex.com/_v/private/typings/linked/v1/vtex.product-summary@2.74.1+build1628605519/public/@types/vtex.product-summary",
```
Comment (victorhmp) [inline react/package.json:52]: Reminder to update this once `vtex.product-summary is released, otherwise, the build will break.

File: react/package.json [hunk 3]
```diff
@@ -41,28 +41,28 @@
...
+    "vtex.render-runtime": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.render-runtime@8.131.0/public/@types/vtex.render-runtime",
     "vtex.responsive-values": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.responsive-values@0.4.2/public/@types/vtex.responsive-values",
     "vtex.rich-text": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.rich-text@0.14.0/public/@types/vtex.rich-text",
     "vtex.search-page-context": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.search-page-context@0.2.0/public/@types/vtex.search-page-context",
-    "vtex.search-result": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.search-result@3.99.1/public/@types/vtex.search-result",
+    "vtex.search-result": "https://specification--motoroladevelopment.myvtex.com/_v/private/typings/linked/v1/vtex.search-result@3.106.1+build1628605526/public/@types/vtex.search-result",
```
Comment (victorhmp) [inline react/package.json:57]: Reminder to update this once `vtex.product-summary is released, otherwise, the build will break.

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 4]
```diff
@@ -126,7 +126,7 @@ or
 }
 ```
 
-> ⚠️ *When set as `skip`, the `simulationBehavior` prop defines that the search data should only be fetched using the store's Cache. Caution: In practice, **this may impact the content displayed on store pages, since the cache storage changes according to user interaction in each page**. In order to understand this prop behavior, take a look at the table below!
+> ⚠️ \*When set as `skip`, the `simulationBehavior` prop defines that the search data should only be fetched using the store's Cache. Caution: In practice, **this may impact the content displayed on store pages, since the cache storage changes according to user interaction in each page**. In order to understand this prop behavior, take a look at the table below!
```
Comment (igorbrasileiro) [inline docs/README.md:129]: This back slash should be here?

File: docs/README.md [hunk 5]
```diff
@@ -135,71 +135,71 @@ or
...
+        "skusFilter": "FIRST_AVAILABLE",
+        "simulationBehavior": "skip"
+      }
+    }
   }
 }
 ```
 
-> ⚠️ *When set as `skip`, the `simulationBehavior` prop defines that the search data should only be fetched using the store's Cache. Caution: In practice, **this may impact the content displayed on store pages, since the cache storage changes according to user interaction in each page**. In order to understand this prop behavior, take a look at the table below!
+> ⚠️ \*When set as `skip`, the `simulationBehavior` prop defines that the search data should only be fetched using the store's Cache. Caution: In practice, **this may impact the content displayed on store pages, since the cache storage changes according to user interaction in each page**. In order to understand this prop behavior, take a look at the table below!
```
Comment (igorbrasileiro) [inline docs/README.md:183]: This back slash should be here?

**Review by igorbrasileiro — APPROVED** [review]
Great job @Git-the-Sanz !! Apply the requests made by @victorhmp and that's it!

File: .eslintrc [hunk 6]
```diff
@@ -1,13 +1,7 @@
...
-      "files": ["*.js", "*.jsx"],
-      "parser": "babel-eslint",
-      "parserOptions": {
-        "sourceType": "module"
-      }
-    }
-  ]
+  "env": {
+    "node": true
+  }
```
[the remark this answers is not in the document]
Comment (Git-the-Sanz) [reply .eslintrc:6]: Yeah! Want me to erase them?

File: docs/README.md (hunk 1 above)
Comment (Git-the-Sanz) [reply docs/README.md:308]: We already talked with the education team, they told us they will create the recipe once the PR is merged. I can come back and update the readme with the specific URL once we have it; what do you think?

File: react/package.json (hunk 2 above)
Comment (Git-the-Sanz) [reply react/package.json:52]: I don't fully grasp how this works. What are the next steps to avoid breaking it?

File: react/package.json (hunk 3 above)
Comment (Git-the-Sanz) [reply react/package.json:57]: Same as the [comment above](https://github.com/vtex-apps/search-result/pull/536#discussion_r690702867) 😰

File: docs/README.md (hunk 4 above)
Comment (Git-the-Sanz) [reply docs/README.md:129]: I used @vtex/prettier-config to format the file. This added it automatically, I bet that there's a rule that adds a backlash when it encounters an asterisk; maybe it thinks it's a comment?

Comment (Git-the-Sanz) [reply docs/README.md:129]: [Indeed!](https://github.com/prettier/prettier/issues/3837)

File: docs/README.md (hunk 5 above)
Comment (Git-the-Sanz) [reply docs/README.md:183]: > [Indeed!](https://github.com/prettier/prettier/issues/3837)

Check this comment

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- `preferredSKU` prop to the `Gallery`.
```
Comment (victorhmp) [inline CHANGELOG.md:11]:
```suggestion
## [Unreleased]
### Added
- `preferredSKU` prop to the `Gallery`.

```

File: docs/README.md [hunk 8]
```diff
@@ -1,11 +1,11 @@
-
-
 📢 Use this project, [contribute](https://github.com/vtex-apps/search-result) to it or open issues to help evolve it using [Store Discussion](https://github.com/vtex-apps/store-discussion).
```
Comment (victorhmp) [inline docs/README.md:1]: Please revert unnecessary changes in this file

File: react/components/GalleryItem.tsx [hunk 9]
```diff
@@ -32,12 +35,13 @@ function GalleryItem({
...
   summary,
   CustomSummary,
+  preferredSKU
 }: GalleryItemProps) {
   const { push } = usePixel()
   const { searchQuery } = useSearchPage()
 
   const product = useMemo(
-    () => ProductSummary.mapCatalogProductToProductSummary(item),
+    () => ProductSummary.mapCatalogProductToProductSummary(item, preferredSKU),
```
Comment (victorhmp) [inline react/components/GalleryItem.tsx:44]: Shouldn't `preferredSKU` be added to the dependency array for this `useMemo` call?

File: react/components/GalleryLayoutItem.tsx [hunk 10]
```diff
@@ -22,12 +25,13 @@ const GalleryLayoutItem: React.FC<GalleryLayoutItemProps> = ({
...
   position,
   listName,
+  preferredSKU
 }) => {
   const { push } = usePixel()
   const { searchQuery } = useSearchPage()
 
   const product = useMemo(
-    () => ProductSummary.mapCatalogProductToProductSummary(item),
+    () => ProductSummary.mapCatalogProductToProductSummary(item, preferredSKU),
```
Comment (victorhmp) [inline react/components/GalleryLayoutItem.tsx:34]: Also here

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/components/GalleryItem.tsx (hunk 9 above)
Comment (Git-the-Sanz) [reply react/components/GalleryItem.tsx:44]: Added

File: react/components/GalleryLayoutItem.tsx (hunk 10 above)
Comment (Git-the-Sanz) [reply react/components/GalleryLayoutItem.tsx:34]: Added

File: docs/README.md (hunk 8 above)
Comment (Git-the-Sanz) [reply docs/README.md]: Reverted the prettified readme

File: CHANGELOG.md (hunk 7 above)
Comment (Git-the-Sanz) [reply CHANGELOG.md]: Commited suggestion

**Review by victorhmp — COMMENTED** [review]
The code LGTM! But I'm still worried about this bit:

> Keep in mind that the default behavior for the initially selected SKU, which selected the first available SKU, has now changed to a Product (field) specification.

Since the default changed, won't this cause unwanted side effects in stores that are using this app?

**Comment by Git-the-Sanz** [comment]
> The code LGTM! But I'm still worried about this bit:
> 
> > Keep in mind that the default behavior for the initially selected SKU, which selected the first available SKU, has now changed to a Product (field) specification.
> 
> Since the default changed, won't this cause unwanted side effects in stores that are using this app?
> 
> The same worry I have with the other PR that also introduces this change.

Maybe we could change the wording. But actually, If you don't implement the `specification` the default behaviour stays the same

**Review by igorbrasileiro — APPROVED** [review]
LGTM, great job @Git-the-Sanz .

**Comment by victorhmp** [comment]
> > The code LGTM! But I'm still worried about this bit:
> > > Keep in mind that the default behavior for the initially selected SKU, which selected the first available SKU, has now changed to a Product (field) specification.
> > 
> > 
> > Since the default changed, won't this cause unwanted side effects in stores that are using this app?
> > The same worry I have with the other PR that also introduces this change.
> 
> Maybe we could change the wording. Because actually, If you don't implement the `specification`, the default behavior stays the same

Ooooh!! That's great then! If that's the case, yep, just changing the wording here solves it 😄 

Could you update it?

File: docs/README.md [hunk 11]
```diff
@@ -302,6 +302,18 @@ To understand how to build your search results with multiple layouts using the `
...
+For `PreferredSKUEnum`:
+
+| Name            | Value             | Description                                        |
+| --------------- | ----------------- | -------------------------------------------------- |
+| First Available | `FIRST_AVAILABLE` | First available SKU in stock found or first SKU without stock. |
+| Last Available  | `LAST_AVAILABLE`  | Last available SKU in stock found or last SKU without stock.  |
+| Cheapest        | `PRICE_ASC`       | Cheapest SKU in stock found or first SKU without stock.        |
+| Most Expensive  | `PRICE_DESC`      | Most expensive SKU in stock found or first SKU without stock.  |
+
+⚠️ Keep in mind that the default behaviour for the initially selected SKU, which selected the first available SKU, has now changed to a Product (field) specification. If the specification doesn't exist, it will use the `preferredSKU` prop as fallback. You can read more about it, and how to implement it in [Recipes](https://vtex.io/docs/recipes/all)
```
Comment (victorhmp) [inline docs/README.md:316]: Please update the wording here so that it doesn't imply that a breaking change was made :)

**Review by victorhmp — CHANGES_REQUESTED** [review]
Code LGTM!
Just the same comment I made in your other PR ;)

File: docs/README.md (hunk 11 above)
Comment (Git-the-Sanz) [reply docs/README.md]: Done!

**Review by Git-the-Sanz — COMMENTED** [review]
Changed Readme

**Review by victorhmp — APPROVED** [review]

**Comment by Git-the-Sanz** [comment]
@icazevedo your review still shows as pending even though it was already addressed. Please approve or remove the request.

**Review by hiagolcm — APPROVED** [review]
The code and the doc looks good and clean. Nice job

**Review by icazevedo — APPROVED** [review]
LGTM

**Review by georgebrindeiro — APPROVED** [review]
Docs LGTM
